ml-agents/mlagents/trainers/tests/test_settings.py

```diff
@@ -289,3 +316,92 @@ def test_env_parameter_structure():
         EnvironmentParameterSettings.structure(
             invalid_curriculum_dict, Dict[str, EnvironmentParameterSettings]
         )
+
+
+def test_exportable_settings():
+    """
+    Test that structuring and unstructuring a RunOptions object results in the same
+    configuration representation.
+    """
+    # Try to enable as many features as possible in this test YAML to hit all the
+    # edge cases. Set as much as possible as non-default values to ensure no flukes.
+    # TODO: Add back in environment_parameters
+    test_yaml = """
+    behaviors:
+        3DBall:
+            trainer_type: sac
+            hyperparameters:
+                learning_rate: 0.0004
+                learning_rate_schedule: constant
+                batch_size: 64
+                buffer_size: 200000
+                buffer_init_steps: 100
+                tau: 0.006
+                steps_per_update: 10.0
+                save_replay_buffer: true
+                init_entcoef: 0.5
+                reward_signal_steps_per_update: 10.0
+            network_settings:
+                normalize: false
+                hidden_units: 256
+                num_layers: 3
+                vis_encode_type: nature_cnn
+                memory:
+                    memory_size: 1288
+                    sequence_length: 12
+            reward_signals:
+                extrinsic:
+                    gamma: 0.999
+                    strength: 1.0
+                curiosity:
+                    gamma: 0.999
+                    strength: 1.0
+            keep_checkpoints: 5
+            max_steps: 500000
+            time_horizon: 1000
+            summary_freq: 12000
+            checkpoint_interval: 1
+            threaded: true
+    env_settings:
+        env_path: test_env_path
+        env_args:
+            - test_env_args1
+            - test_env_args2
+        base_port: 12345
+        num_envs: 8
+        seed: 12345
+    engine_settings:
+        width: 12345
+        height: 12345
+        quality_level: 12345
+        time_scale: 12345
+        target_frame_rate: 12345
+        capture_frame_rate: 12345
+        no_graphics: true
+    checkpoint_settings:
+        run_id: test_run_id
+        initialize_from: test_directory
+        load_model: false
+        resume: true
+        force: true
+        train_model: false
+        inference: false
+    debug: true
+    """
+    loaded_yaml = yaml.safe_load(test_yaml)
+    run_options = RunOptions.from_dict(yaml.safe_load(test_yaml))
+    dict_export = run_options.as_dict()
+    check_dict_is_at_least(loaded_yaml, dict_export)
```

Comment:
Do we really need the yaml and the exported dict to be identical ? Or is it enough that the dict export gives the same RunOptions when reloaded ?

Reply:
The reason I ask is because in my PR, I will probably rewrite the parameter samplers as curriculums of 1 lesson

Reply:
I guess we don't. We can also add `environment_parameters` as an exclusion in the first call - I'm OK with either.